### DISCUSSION TITLE
Blocking the receipt of achievements

### DISCUSSION THREAD

**Discussion by dalandis** [Zrips/CMI#4062] [Feature Requests]
**Description of feature/improvement request:**
Blocking the receipt of achievements

---
### Detailed explanation of what it should do, when and in what situation: ###
Blocking achievements if the player is in creative or spectator or vanish.

### Basic explanation as to why this feature is needed and/or useful to have: ###
For servers with multiworlds, to block achievements in the creative world.
If the moderator follows the player and gets the achievement, just like the player. Especially if BlazeandCave's Advancements Pack is installed.

**Comment by dalandis** [comment]
I would like to be able not only to block announcements of achievements. Block exactly the very receipt of achievements.

**Comment by Polda18** [comment]
This is very much needed feature. I got my cover blown by this very exact thing. Block announcements of achievements whenever I'm in vanish, and on top of that and independently of that, block reception of achievements when in creative or spectator mode. Bedrock Edition Minecraft disables achievements entirely when a player enables cheats in the world, which means access to creative and op commands. Java Edition for some reason still gives you achievements even when you are in creative mode. I'm following a player suspected from cheating and suddenly my cover is blown by the stupid achievements.

I would actually make it an option in vanishedit. Make it supressed by default, but let players enable it if they like.
